contracts/SPGNFT.sol

```diff
@@ -207,25 +217,54 @@ contract SPGNFT is ISPGNFT, ERC721URIStorageUpgradeable, AccessControlUpgradeabl
     /// @notice Mints an NFT from the collection. Only callable by the minter role.
     /// @param to The address of the recipient of the minted NFT.
     /// @param nftMetadataURI OPTIONAL. The URI of the desired metadata for the newly minted NFT.
-    /// @return tokenId The ID of the minted NFT.
-    function mint(address to, string calldata nftMetadataURI) public virtual returns (uint256 tokenId) {
+    /// @param nftMetadataHash OPTIONAL. A bytes32 hash of the NFT's metadata.
+    /// This metadata is accessible via the NFT's tokenURI.
+    /// @param dedup Set to true to enable checking for duplicate metadata hashes.
+    /// If a duplicate is found, the function will return the existing token ID instead of minting a new one.
+    /// @return tokenId The token ID of the minted NFT with the given metadata hash.
+    /// @return deduped true if dedup is enabled and the metadata hash is already used, false otherwise.
+    function mint(
+        address to,
+        string calldata nftMetadataURI,
+        bytes32 nftMetadataHash,
+        bool dedup
+    ) public virtual returns (uint256 tokenId, bool deduped) {
```

Comment:
It would be better revert an error if found the nftMetdataHash already used to mint token before. the caller contract can try/catch the error if needed.

Reply:
Resolved in [9f4a11e](https://github.com/storyprotocol/protocol-periphery-v1/pull/108/commits/9f4a11e1ec434e8a8890a0f651391d31961bae3c)